clang analyzer failing

pip is unable to upgrade typing_extensions module because it is installed from a deb package:
```
Installing collected packages: thrift, jsonpath-ng, typing-extensions, types-PyYAML, smmap, setuptools, semver, psutil, portalocker, pillow, numpy, Mako, lxml, kiwisolver, jinja2, greenlet, fonttools, dill, cycler, Authlib, sqlalchemy, python-docx, multiprocess, gitdb, contourpy, matplotlib, gitpython, alembic, sarif-tools, codechecker
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.10.0
ERROR: Cannot uninstall typing_extensions 4.10.0, RECORD file not found. Hint: The package was installed by debian.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/clang-analyzer.yml
+++ b/.github/workflows/clang-analyzer.yml
@@ -71,10 +71,15 @@ jobs:
     - name: Delete Proton
       run: rm -rf qpid-proton
 
+      # To make sure that pip can upgrade typing-extensions
+    - name: Prepare compilation database
+      run: sudo apt remove -y python3-typing-extensions
+      continue-on-error: true
+
     # https://clang-analyzer.llvm.org/command-line.html
     - name: Create compilation database
       run: |
-        PIP_BREAK_SYSTEM_PACKAGES=1 sudo pip3 install codechecker
+        PIP_BREAK_SYSTEM_PACKAGES=1 sudo --preserve-env=PIP_BREAK_SYSTEM_PACKAGES pip3 install codechecker
         # disable IPO to make compilation faster
         cmake -S . -B build -DBUILD_TESTING=OFF -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=OFF -DQD_ENABLE_ASSERTIONS=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DENABLE_WARNING_ERROR=OFF
         # perform build so that generated sources get generated

